Squadrosu.Framework/Piece.cs

```diff
@@ -1,20 +1,169 @@
+// Copyright (c) SquadroCorp
+// This file is part of Squadrosu!.
+// Squadrosu! is licensed under the GPL v3. See LICENSE.md for details.
+
+using System;
+using System.Dynamic;
+using System.Runtime.CompilerServices;
+
 namespace Squadrosu.Framework;
 
-public class Piece 
+public class Piece
 {
     /// <summary>
-    /// To which player is the current piece.
+    /// To which player is the current piece
     /// </summary>
-    private Player Player;
+    public readonly Player Player;
     /// <summary>
     /// Direction of the piece
     /// </summary>
-    private int Direction { get; set; }
+    public Direction Direction { get; private set; }
     /// <summary>
     /// Position in the row
     /// </summary>
-    private int Position { get; set; }
+    public int Position { get; private set; }
+    /// <summary>
+    /// Line number
+    /// </summary>
+    public readonly int LineNumber;
+
+    public Piece(Player Player, int LineNumber)
+    {
+        Direction = Direction.Forward;
+        Position = 0;
+        this.LineNumber = LineNumber;
+        this.Player = Player;
+    }
+
+    /// <summary>
+    /// The piece returns to the initial position of the direction
+    /// </summary>
+    public void Reset()
+    {
+        this.Position = (this.Direction == Direction.Forward) ? 0 : 6;
+    }
```

Comment:
Je m'attendais pas à ça, mais je vois en quoi c'est utile...
Mais pour le coup je pense que ça devrait être une autre fonction.
```suggestion
    public void PlaceOnEdge()
    {
        Position = Direction == Direction.Forward ? 0 : 6;
    }

    public void Reset()
    {
        Direction = Direction.Forward;
        PlaceOnEdge();
    }
```

Reply:
C'est bon